eos/utils/interpolation_TEST.cc

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2022 Philip Lüghausen
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+#include <interpolation.hh>
+#include <eos/utils/parameters.hh>
+
+using namespace test;
+using namespace eos;
+
+class InterpolationTest :
+    public TestCase
+{
+    public:
+        InterpolationTest() :
+            TestCase("interpolation_test")
+        {
+        }
+
+        virtual void run() const
+        {
+            {
+                // standard initialisation
+                CSplineInterpolation interp = CSplineInterpolation({0.0, 1.0, 2.0}, {0.0, 1.0, 2.0});
+                TEST_CHECK_NEARLY_EQUAL(interp(0.5), 0.5, 1e-14);
+            }
+            {
+                // uniform initialization syntax
+                CSplineInterpolation interp = {
+                    {0.0, 1.0, 2.0},
+                    {0.0, 1.0, 2.0}
+                };
+                TEST_CHECK_NEARLY_EQUAL(interp(0.5), 0.5, 1e-14);
+            }
+            {
+                // evaluate outside of data range: must throw
+                CSplineInterpolation interp = {
+                    {0.0, 1.0, 2.0},
+                    {0.0, 1.0, 2.0}
+                };
+                TEST_CHECK_THROWS(GSLError, interp(3.0));
+            }
+            {
+                // map domain of x
+                CSplineInterpolation interp = {
+                    {0.0, 1.0, 2.0},
+                    {0.0, 1.0, 2.0},
+                    [](const double & x) {return 2. * x;}
```

Comment:
Ahh, I see. This is a bit overkill. This class does not care how it is used or that you map an e.g. infinite domain to a finite one as in your application. Let's please keep the mapping an implementation detail of the user of ``CSplineInterpolation``.